public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -118,19 +118,18 @@ export class TempoDatasource extends DataSourceWithBackend<TempoQuery, TempoJson
   }
 
   async testDatasource(): Promise<any> {
-    // to test Tempo we send a dummy traceID and verify Tempo answers with 'trace not found'
-    const response = await super.query({ targets: [{ query: '0' }] } as any).toPromise();
-
-    const errorMessage = response.error?.message;
-    if (
-      errorMessage &&
-      errorMessage.startsWith('failed to get trace') &&
-      errorMessage.endsWith('trace not found in Tempo')
-    ) {
+    const options: BackendSrvRequest = {
+      headers: {},
+      method: 'GET',
+      url: `${this.instanceSettings.url}/api/echo`,
+    };
+    const response = await getBackendSrv().fetch<any>(options).toPromise();
+
+    if (response.ok) {
       return { status: 'success', message: 'Data source is working' };
     }
 
-    return { status: 'error', message: 'Data source is not working' + (errorMessage ? `: ${errorMessage}` : '') };
+    return { status: 'error', message: 'Data source is not working' };
```

Comment:
I think this line might not be reachable. [BackendSrv.toFailureStream ](https://github.com/grafana/grafana/blob/main/public/app/core/services/backend_srv.ts#L201)filters all responses with `response.ok === false` and for them throws an error (with data processed in [processRequestError()](https://github.com/grafana/grafana/blob/main/public/app/core/services/backend_srv.ts#L307)). The error is being handled in [testDataSource action](https://github.com/grafana/grafana/blob/main/public/app/features/datasources/state/actions.ts#L104) so it should be safe to remove this line unless you need a specific error message (but then it'd need to be in a `catch`).

Reply:
You're right, when testing I was only seeing other error messages which must be coming from that function. I'll remove it.